### PR TITLE
Add field source to GitProjectModel

### PR DESCRIPTION
**Pull request by sakalosj**
Agreed on updating GitProjectModel:

* field: instance_url
* content: git service url (str)

### CODE REVIEW

File: packit_service/models.py [hunk 1]
```diff
@@ -131,6 +132,11 @@ class GitProjectModel(Base):
     # Example: https://github.com/packit/hello-world.git
     https_url = Column(String)
     project_url = Column(String)
+    source = Column(String, nullable=False)
```
Comment (lachmanfrantisek) [inline packit_service/models.py:135]: What about calling it `instance_url` like in other places across the code and in OGR?

The thing that is (probably) not consistent is if we add protocol or not.

File: alembic/versions/e35bb9ce7313_added_source_field_to_gitproject.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+depends_on = None
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    op.add_column("git_projects", sa.Column("source", sa.String(), nullable=True))
+    for git_project in session.query(GitProjectModel):
+        git_project.source = urlparse(git_project.project_url).netloc
+    session.commit()
```
Comment (lachmanfrantisek) [inline alembic/versions/e35bb9ce7313_added_source_field_to_gitproject.py:29]: Nice!

**Review by lachmanfrantisek — COMMENTED** [review]
Nice! Do you want to work on the usage in another PR?

(e.g. simplifying/getting rid of `__choose_project`)

File: packit_service/models.py (hunk 1 above)
Comment (TomasTomecek) [reply packit_service/models.py]: agreed with the rename, "source" can literally mean anything

File: alembic/versions/e35bb9ce7313_added_instance_url_field_to_gitproject.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    op.add_column("git_projects", sa.Column("instance_url", sa.String(), nullable=True))
+    for git_project in session.query(GitProjectModel):
+        git_project.instance_url = urlparse(git_project.project_url).netloc
```
Comment (lachmanfrantisek) [inline alembic/versions/e35bb9ce7313_added_instance_url_field_to_gitproject.py:28]: I am only wondering if `.hostname` (without protocol) will not be more useful. What do you think?

```suggestion
        git_project.instance_url = urlparse(git_project.project_url).hostname
```

Comment (sakalosj) [reply alembic/versions/e35bb9ce7313_added_instance_url_field_to_gitproject.py]: fixed

File: packit_service/models.py (hunk 1 above)
Comment (sakalosj) [reply packit_service/models.py]: fixed

**Review by TomasTomecek — APPROVED** [review]
LGTM

though I'm no longer familiar with p-s' codebase so I'd suggest to account for other reviews as well

File: packit_service/models.py [hunk 4]
```diff
@@ -131,6 +132,11 @@ class GitProjectModel(Base):
     # Example: https://github.com/packit/hello-world.git
     https_url = Column(String)
     project_url = Column(String)
+    instance_url = Column(String, nullable=False)
+
+    @orm.reconstructor
+    def __init__(self):
+        self.instance_url = urlparse(self.project_url).netloc
```
Comment (lachmanfrantisek) [inline packit_service/models.py:139]: Sorry, forgot to mention also this one:

```suggestion
        self.instance_url = urlparse(self.project_url).hostname
```

Comment (sakalosj) [reply packit_service/models.py]: np, was bit tired later yesterday, fixed now

**Comment by sakalosj** [comment]
> Nice! Do you want to work on the usage in another PR?
> 
> (e.g. simplifying/getting rid of `__choose_project`)

Yes, I can check it, but not sure if will have enough time for it.

**Pull request merged**